signal/cmd/run.go

```diff
@@ -141,24 +112,13 @@ var (
 			proto.RegisterSignalExchangeServer(grpcServer, srv)
 
 			grpcRootHandler := grpcHandlerFunc(grpcServer)
+
+			startServer(certManager, grpcRootHandler)
```

Comment:
It would be more understandable if you put the `if certManager != nil {` outside of the start function. Because now has serve function inside the start and outside too and is not trivial when will be used for.